From c51f40b1822a6f8758311e64f21a7577b75715b8 Mon Sep 17 00:00:00 2001
From: Alastair D'Silva <alastair@d-silva.org>
Date: Sat, 15 Aug 2026 20:21:12 +1000
Subject: [PATCH 08/23] configs: sunxi: Add defconfig for Orange Pi 4 Pro

Add defconfig for Xunlong Orange Pi 4 Pro based on Allwinner A733 SoC.
Enables LPDDR4 DRAM, MMC0/MMC2, DWC ETH QOS with YT8531 PHY, UART0,
R_I2C PMIC, and USB EHCI/OHCI.

Follows the upstream orangepi_4a_defconfig pattern by Andre Przywara.

Signed-off-by: Alastair D'Silva <alastair@d-silva.org>
Assisted-by: Antigravity <antigravity@google.com>
---
 configs/orangepi_4_pro_defconfig | 35 ++++++++++++++++++++++++++++++++
 1 file changed, 35 insertions(+)
 create mode 100644 configs/orangepi_4_pro_defconfig

diff --git a/configs/orangepi_4_pro_defconfig b/configs/orangepi_4_pro_defconfig
new file mode 100644
index 00000000000..0fbc684ddef
--- /dev/null
+++ b/configs/orangepi_4_pro_defconfig
@@ -0,0 +1,35 @@
+CONFIG_ARM=y
+CONFIG_ARCH_SUNXI=y
+CONFIG_DEFAULT_DEVICE_TREE="allwinner/sun60i-a733-orangepi-4-pro"
+CONFIG_SPL=y
+CONFIG_DRAM_SUNXI_DX_ODT=0x07070707
+CONFIG_DRAM_SUNXI_DX_DRI=0x0d0d0d0d
+CONFIG_DRAM_SUNXI_CA_DRI=0x0e0e
+CONFIG_DRAM_SUNXI_ODT_EN=0x84848484
+CONFIG_DRAM_SUNXI_TPR0=0x80808080
+CONFIG_DRAM_SUNXI_TPR1=0x06060606
+CONFIG_DRAM_SUNXI_TPR2=0x5f090503
+CONFIG_DRAM_SUNXI_TPR6=0x38000000
+CONFIG_DRAM_SUNXI_TPR10=0x802f3333
+CONFIG_DRAM_SUNXI_TPR11=0xc6c4c2c0
+CONFIG_DRAM_SUNXI_TPR12=0x3a373233
+CONFIG_MACH_SUN60I_A733=y
+CONFIG_SUNXI_DRAM_A733_LPDDR4=y
+CONFIG_MMC_SUNXI_SLOT_EXTRA=2
+CONFIG_R_I2C_ENABLE=y
+CONFIG_SPL_I2C=y
+CONFIG_SPL_SYS_I2C_LEGACY=y
+CONFIG_SYS_I2C_MVTWSI=y
+CONFIG_SYS_I2C_SLAVE=0x7f
+CONFIG_SYS_I2C_SPEED=400000
+CONFIG_SUPPORT_EMMC_BOOT=y
+CONFIG_USB_EHCI_HCD=y
+CONFIG_USB_OHCI_HCD=y
+CONFIG_DWC_ETH_QOS=y
+CONFIG_DWC_ETH_QOS_SUNXI=y
+CONFIG_PHY_MOTORCOMM=y
+CONFIG_PHY_GIGE=y
+CONFIG_CMD_MII=y
+CONFIG_CMD_MDIO=y
+CONFIG_CMD_PING=y
+CONFIG_CMD_DHCP=y
-- 
2.53.0

